freespace2/freespace.cpp

```diff
@@ -1376,11 +1389,15 @@ bool game_start_mission()
 
 		return false;
 	}
-	load_mission_load = (uint) (time(nullptr) - load_mission_load);
 
-	// free up memory from parsing the mission
-	extern void stop_parse();
-	stop_parse();
+	// Since we just freed up memory, now is probably a good time to run the Lua garbage collector.
+	// This is also after the preliminary checks are complete but before things start getting paged in.
+	auto L = Script_system.GetLuaSession();
+	if (L != nullptr)
+	{
+		game_busy(NOX("** running Lua garbage collector **"));
+		luaC_fullgc(L);
+	}
```

Comment:
Is it worth adding an `else` clause where we log that we were unable to get the Lua session?

If `L` were `nullptr`, would that mean something's wrong? Or not necessarily?

Reply:
I don't know, TBH.  It might mean that scripting isn't supported, or that the engine was built without scripting support.  It might not actually be *possible* for `L` to be `nullptr` here, but I figured better safe than sorry.

Reply:
Oh I absolutely agree that we should defensively check for `nullptr`. I was just wondering whether it'd be worthwhile to log a debug message if `L` *is* `nullptr`, e.g.,
```
Attempted to run Lua GC after parsing the mission but Lua session is null
```
but it's not important.

Reply:
I don't think that's necessary.  If `L` is valid, the garbage collector should be run; if not, the gc is irrelevant.